src/pudl/analysis/state_demand.py

```diff
@@ -390,17 +406,23 @@ def impute_ferc714_hourly_demand_matrix(df: pd.DataFrame) -> pd.DataFrame:
     Args:
         df: FERC 714 hourly demand matrix,
           as described in :func:`load_ferc714_hourly_demand_matrix`.
+        years: list of years to input
 
     Returns:
         Copy of `df` with imputed values.
     """
     results = []
-    for year, gdf in df.groupby(df.index.year):
-        logger.info(f"Imputing year {year}")
-        keep = df.columns[~gdf.isnull().all()]
-        tsi = pudl.analysis.timeseries_cleaning.Timeseries(gdf[keep])
-        result = tsi.to_dataframe(tsi.impute(method="tnn"), copy=False)
-        results.append(result)
+    # sort here and then don't sort in the groupby so we can process
+    # the newer years of data first. This is so we can see early if
+    # new data causes any failures.
+    df = df.sort_index(ascending=False)
+    for year, gdf in df.groupby(df.index.year, sort=False):
+        if year in years:
+            logger.info(f"Imputing year {year}")
+            keep = df.columns[~gdf.isnull().all()]
+            tsi = pudl.analysis.timeseries_cleaning.Timeseries(gdf[keep])
+            result = tsi.to_dataframe(tsi.impute(method="tnn"), copy=False)
+            results.append(result)
```

Comment:
i did two things here:
- make the newer years run first - bc this takes 15 minutes to run omigosh and presumably the new failures will come in the new years.
- restrict the years that this impute is run on - bc the xbrl data introduced "Hour 24" records that are the first second of Jan 1 of the following year & this impute needs... well at least more than one record for the whole year

Reply:
hmm ok! I thought we fixed the weird hour data for XBRL. I remember the issue was with Hour 0 vs. Hour 24 etc. but maybe we just made then all Hour 24. Should we correct it so it's not the first second of the new year?